src/components/storefront/TeamBanner/TeamBanner.js

```diff
@@ -0,0 +1,38 @@
+import { Flex, Spinner, Text, useDisclosure } from '@chakra-ui/react';
+import { InfoOutlineIcon } from '@chakra-ui/icons';
+import NoTeamAssociationModal from '../NoTeamAssociationModal/NoTeamAssociationModal';
+
+const TeamBanner = props => {
+  const { loading, error, data } = props;
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  return (
+    <Flex
+      onClick={!loading && (error || !data) ? onOpen : null}
+      cursor={!loading && (error || !data) ? 'pointer' : null}
+      w="100%"
+      h="34px"
+      justify="center"
+      align="center"
+      bg={loading || data ? 'brand.lightgray' : 'brand.red'}
+    >
+      <Text color={loading || data ? 'black' : 'white'} textTransform="uppercase" fontWeight="bold">
+        {loading ? (
+          <Spinner size="sm" />
+        ) : data ? (
+          `Team ${data.getTeam.name}`
+        ) : (
+          'Please note that your purchase will not be attributed to a team'
+        )}
```

Comment:
Put in a spinner instead of "Loading..." text. Also refactored to a ternary statement for better readability